scripts/lib/CIME/XML/env_batch.py

```diff
@@ -185,32 +186,34 @@ def make_batch_script(self, input_template, job, case, total_tasks, tasks_per_no
             fd.write(output_text)
         os.chmod(job, os.stat(job).st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
 
-    def set_job_defaults(self, batch_jobs, pesize=None, walltime=None, force_queue=None, allow_walltime_override=False):
+    def set_job_defaults(self, batch_jobs, pesize=None, num_nodes=None, walltime=None, force_queue=None, allow_walltime_override=False):
         if self.batchtype is None:
             self.batchtype = self.get_batch_system_type()
 
         if self.batchtype == 'none':
             return
 
         for job, jsect in batch_jobs:
-            task_count = jsect["task_count"]
-            if task_count is None or task_count == "default":
+            node_count = jsect["node_count"]
+            if node_count is None or node_count == "default":
                 task_count = pesize
+                node_count = num_nodes
             else:
-                task_count = int(task_count)
+                task_count = 1
+                node_count = int(node_count)
```

Comment:
Although we have only used a task_count of 1 here, it was never intended that that be the only choice.